yogstation/code/modules/admin/verbs/adminvox.dm

```diff
@@ -0,0 +1,59 @@
+//Check over there to see what's different, if this ever breaks.
+  set category = "Fun"
+  set name = "Force AI Vox"
+  set hidden = 0
+
+  if(!check_rights(R_ADMIN))
+    return
+
+  var/message = input(src, "WARNING: Misuse of this verb can result in you being yelled at by Ross.", "Announcement") as text
+//^ Remember to replace "Ross" with whoever owns the server in 20,000 years after Ross either dies of natural causes or is assassinated by Oakboscage
+  var/voxType = input(src, "Male or female VOX?", "VOX-gender") in list("male", "female") //yogs - male vox
+
+  if(!message)
+    return
+
+  var/list/words = splittext(trim(message), " ")//Turns the received text into an array of words
+  var/list/incorrect_words = list()//Stores all the words that we can't say, so we can complain about it later
+
+  if(words.len > 30)
+  // While normally the length of an array is something you can only read and never write,
+  //in DM, len is actually how you manage the size of the array. Editing it to value to like this is actually totally doable.
+    words.len = 30
+
+  if(voxType == "female") // The if for this is OUTSIDE the for-loop, for optimization.
+  //Putting it inside makes it check for it every single damned time it parses a word,
+  //Which might be super shitty if someone removes the length limit above at some point.
+    for(var/word in words) // For each word
+      word = lowertext(trim(word)) // We store the words as lowercase, so lowercase the word and trim off any weirdness like newlines
+      if(!word) // If we accidentally captured a space or something weird like that
+        words -= word // Scratch this one off the list
+        continue // and skip over it
+      if(!GLOB.vox_sounds[word])
+        incorrect_words += word
+  else if(voxType == "male") // If we're doing the yog-ly male AI vox voice
+    for(var/word in words)
+      word = lowertext(trim(word))
+      if(!word)
+        words -= word
+        continue
+      if(!GLOB.vox_sounds_male[word])
+        incorrect_words += word
+  else
+    to_chat(src,"<span class='notice'>Unknown or unsupported vox type. Yell at a coder about this.</span>")
+
+  if(incorrect_words.len)
+    to_chat(src, "<span class='notice'>These words are not available on the announcement system: [english_list(incorrect_words)].</span>")
+    return
+
+  log_admin("[key_name(src)] made an admin AI vocal announcement with the following message: [message].")
+  message_admins("[key_name_admin(src)] has forced an AI vox announcement.")
+  if(src.mob) // If the admin has a mob, whether it's a ghost or whatever
+    for(var/word in words) // Then play it
+      play_vox_word(word, src.mob.loc.z, null, voxType) //yogs - male vox
+  else // If, for some ungodly reason, the admin lacks a mob
+    for(var/word in words) // Then play it on the station's Z-level
+    //TODO: Make a define for the station's z-level
+      play_vox_word(word, 2, null, voxType) //yogs - male vox
```

Comment:
```
var/z_level = 2
if(src.mob && src.mob.loc)
    z_level = src.mob.loc.z
```

Reply:
In another episode of Ling hating there being 4 lines of code in something instead of 2

Reply:
```
Okay, so, since this is apparently enough to merit this PR being declared unmergable, here's a couple more points.
```
It wasn't only because of this but okay. :)


```
Your change would take a 6-line thing and make it 5 lines, which is not an absolutely revolutionary change to begin with.
```
What does linecount have to do with this? When did I ever say this was about linecount? I proposed a solution that looks cleaner and that allows you to instantly see that the default z-level is 2. Any person that wants to change it in the future will be able to instantly recognize and edit it. If L55 wasn't given, you wouldn't be able to tell the second argument of ``play_vox_word`` is the z-level. My proposed solution eliminates this problem by having a variable named ``z_level`` (or something alike) so any future coders can instantly see that the second argument is the z-level. The reason I requested this change is because of readability and maintainability, not "muh linecount".

```
Said change (to reduce its line usage by one) would introduce the use of a dummy variable for the garbage collector to clean up. Not only is memory allocation one of the most expensive operations available (and so a violation of style and a waste of cache space)
```
Oh no, the GC has to clean up 1 extra variable 😱. How will we ever cope with this performance degradation and memory waste? We better elimate all local variables to unlock the full potential of BYOND and get maximum framerates. If we assume BYOND uses a proper implementation, the variable (wether its a straight up float or the float is inside a class), it will most likely be allocated on the stack. If you're complaining about allocating things on the stack, geez. Even if it wasn't allocated on the stack, the performance you lose by allocating a class in ram is still negligible.

Regarding the cache space, the program probably wouldn't even need to grab the variable from RAM. It was created a few instructions before so it will most likely still be in a register/on the stack. Worst case, you have 1 cache miss. Afterwards, the variable isn't used anymore and the CPU overrides the unused variable with something else, effectively wasting no cache memory at all.


```
but we also are not 100% sure that Byond even has a working GC, especially when you consider how the game tends to degrade in performance and memory usage the longer the round goes on.
```
That poor variable that only gets created once admins use some niche button. I'm sure we're all gonna cry over all those potentially wasted bytes. And I mean literally bytes, BYOND uses a 32-bit float to represent numbers. Add some extra bytes used to track usage count and stuff and you're looking at like 16 bytes every time this is used (assuming the variable isn't cleaned/reused), worst case.

```
even at a cost to what those style guides are trying to achieve
```
Please enlighten me, what are these style guide trying to achieve? Because clearly I don't know.